api/controllers/v0.3/ILSHelper.js

```diff
@@ -0,0 +1,390 @@
+    }
+
+    // False if patron provides a home address that is not residential
+    // False if patron does not have a recognized name
+    // False if patron policy is not the default (:simplye)
+    return (
+      patron.address.isResidential &&
+      patron.hasValidName &&
+      patron.policy.isDefault
+    );
+  }
+
+  patronFields(patron) {
+    let fieldData = this.formattedPatronData(patron);
+
+    let fields = [
+      fieldData["name"],
+      fieldData["address"],
+      fieldData["ptype"],
+      fieldData["username"],
+      fieldData["pin"],
+      fieldData["ecommunications_pref"],
+      // notice_pref_as_field,
+      // home_lib_as_field,
+      patron.policy.agency,
+    ];
+
+    if (patron.worksInCity) {
+      fields.push(fieldData["workAddress"]);
+      fields.push(note_as_field);
+    }
+
+    if (this.longExpiration(patron)) {
+      fields.push(this.longExpirationAsField(patron));
+    } else {
+      patron.set_temporary();
+      fields.push(this.shortExpirationAsField(patron));
+    }
+
+    // Add required/optional fields according to policy.
+    if (patron.barcode) {
+      fields.push(fieldData["barcode"]);
+    }
+    if (patron.email) {
+      fields.push(fieldData["email"]);
+    }
+    if (patron.birthdate) {
+      fields.push(fieldData["birthdate"]);
+    }
+
+    return {
+      patronFields: {
+        patron_field: fields,
+      },
+      patronID: patron.patronId,
+    };
+  }
+
+  blankField() {
+    return { fieldTag: "", marcTag: 0, value: "", value_is_binary: false };
+  }
+
+  customField(fieldTag, value, marcTag = 0) {
+    let field = this.blankField();
+    field["field_tag"] = fieldTag;
+    field["value"] = value;
+    field["marc_tag"] = marcTag;
+    return field;
+  }
+
+  expirationAsField(time) {
+    // TODO convert date object
+    let expiration = time.strftime("%Y%m%d000000");
+    return this.customField(
+      IlsHelper.EXPIRATION_FIELDTAG,
+      expiration,
+      IlsHelper.STRING_MARCTAG
+    );
+  }
+
+  longExpirationAsField(patron) {
+    let timeNow = Date.now();
+    return this.expirationAsField(
+      `${timeNow}${parseInt(patron.policy.card_type["standard"], 10)}`
+    );
+  }
+
+  shortExpirationAsField(patron) {
+    let timeNow = Date.now();
+    return this.expirationAsField(
+      `${timeNow}${parseInt(patron.policy.card_type["temporary"], 10)}`
+    );
+  }
+
+  // Internal: Create dynamic field-creation method.
+  //
+  // field - A String representing the base field name. <field>_FIELDTAG
+  //         must be set in IlsHelper. Can have any capitalization.
+  //
+  // value - A primitive type Object. May be nil. If nil, DEFAULT_<field>
+  //         constant must be set in IlsHelper
+  //
+  // If the field requires the STRING_MARCTAG, it must be included in the
+  // IlsHelper.WITH_MARK_TAG array.
+  //
+  // Signature
+  //
+  //   <field>_as_field([value])
+  //
+  // field - a field name
+  //
+  // Returns a Hash representing the custom field.
```

Comment:
I assume this was referring to a Ruby hash (i.e. a JS object)...?

Reply:
Ah, yes. Those I'll update by removing.